26_CentralLimitBoxBlur/main.cpp

```diff
@@ -0,0 +1,339 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+
+// I've moved out a tiny part of this example into a shared header for reuse, please open and read it.
+#include "../common/MonoDeviceApplication.hpp"
+#include "../common/MonoAssetManagerAndBuiltinResourceApplication.hpp"
+
+#include <nbl/builtin/hlsl/blur/common.hlsl>
+
+#include "CArchive.h"
+
+using namespace nbl;
+using namespace core;
+using namespace system;
+using namespace asset;
+using namespace video;
+
+#define _NBL_PLATFORM_WINDOWS_
+
+class BoxBlurDemo final : public examples::MonoDeviceApplication, public examples::MonoAssetManagerAndBuiltinResourceApplication
+{
+	using device_base_t = examples::MonoDeviceApplication;
+	using asset_base_t = examples::MonoAssetManagerAndBuiltinResourceApplication;
+
+public:
+	BoxBlurDemo( 
+		const path& _localInputCWD, 
+		const path& _localOutputCWD, 
+		const path& _sharedInputCWD, 
+		const path& _sharedOutputCWD 
+	) : system::IApplicationFramework( _localInputCWD, _localOutputCWD, _sharedInputCWD, _sharedOutputCWD )
+	{}
+	
+	bool onAppInitialized( smart_refctd_ptr<ISystem>&& system ) override
+	{
+		// Remember to call the base class initialization!
+		if( !device_base_t::onAppInitialized( std::move( system ) ) )
+		{
+			return false;
+		}
+		if( !asset_base_t::onAppInitialized( std::move( system ) ) )
```

Comment:
you can't move the same smart pointer twice, you need to copy it, see a different example